src/index.ts

```diff
@@ -55,6 +58,19 @@ export class Steveo implements ISteveo {
     this.pool = build(this.config.workerConfig);
     this.events = this.registry.events;
     this.hooks = hooks;
+
+    // Add a termination check when no hooks are present
+    if (!this.hooks?.healthCheck && !this.hooks?.terminationCheck) {
+      for (const signal of ['SIGTERM', 'SIGINT'] as const) {
+        process.on(signal, async () => {
+          this.logger.info(
+            `Received ${signal} -- Disconnecting and terminating`
+          );
+          setTimeout(() => process.exit(0), 100);
```

Comment:
This should not be calling `process.exit`! Imo we should only be providing the abstraction to cleanly terminate connections via `disconnect` and leave callers to manage their own shutdowns

Reply:
makes sense.

Reply:
@rrsmith https://github.com/ordermentum/steveo/blob/45b1e87e37457d2ef2ebc61920d615c9a150ac2a/src/runner/kafka.ts#L116
added a process exit here, where it can't recover after disconnection.

Reply:
Again - library code shouldn't be responsible for this. Provide hooks so callers can see what is happening and take the actions appropriate to them.